AmplifyPlugins/Analytics/AWSPinpointAnalyticsPlugin/LocalStorage/SQLiteLocalStorageAdapter.swift

```diff
@@ -108,3 +112,19 @@ final class SQLiteLocalStorageAdapter: SQLStorageProtocol {
 }
 
 extension SQLiteLocalStorageAdapter: DefaultLogger { }
+
+extension URL: DefaultLogger {
+    var attributes: [FileAttributeKey : Any]? {
```

Comment:
I see how this is convenient, but it's also relying on a hard dependency to `FileManager` that can't be mocked during unit testing.

I suggest that we instead receive an instance of the existing `FileManagerBehaviour` in`SQLiteLocalStorageAdapter.init` that defaults to `FileManager.default`, and then expand that protocol to have the method you need.

Something like:
```swift
protocol FileManagerBehaviour {
    // ...
    func fileSize(for url: URL) -> Int
}

extension FileManager:  FileManagerBehaviour {
    // ...
    func fileSize(for url: URL) -> Int {
        // Grab the size from the attributes
    }
}

class SQLiteLocalStorageAdapter {
    // ...
    let fileManager: FileManagerBehaviour

    init(fileManager: FileManagerBehaviour = FileManager.default) { // + other arguments
        // ...
    }

    var bytesUsedOnDisk: Byte {
        return fileManager.fileSize(for: dbFilePath)
    }
}

```

Once you have a local variable, you can also completely remove all the other references to `FileManager.default`  in the  `SQLiteLocalStorageAdapter` class :)

Reply:
Love the suggestion, code has been updated.